src/hooks/use-focus.ts

```diff
@@ -0,0 +1,40 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+type OnFocusChangeHandler = (focused: boolean) => unknown;
+
+export default function useFocus<E extends HTMLElement>(
+  target?: React.RefObject<E> | null,
+  onFocusChange?: OnFocusChangeHandler
+) {
+  const [focused, setFocused] = useState(() => {
+    return window.document.activeElement != null;
```

Comment:
Does this work even in SSR? I think we could face an error with window being undefined but tell me if you encountered any error

Reply:
Haven't had issues during testing

Reply:
```suggestion
    return window?.document.activeElement != null;
```
Just in case

Reply:
yeah you're right
```suggestion
    return typeof window !== undefined ? window.document.activeElement != null: true;
```